### ISSUE TITLE
Fail gracefully if parsl is installed

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-server#346]
Right now, we rely on a ModuleNotFoundError during the import of available backends, but we get a NotImplementedError.

**Comment by jacopo-exact** [comment]
## TL;DR
This was actually a design choice: the server should fail to start and fail epically if a dependnecy is missing.

## Explanation

Fractal has a series of optional dependencies, which get installed using the extra syntax of `pip` (e.g., `pip install fractal-server[slurm]`). However, relative to this example, if the sysadmin installs Fractal and sets `FRACTAL_RUNNER_BACKEND=slurm`, the dependency is effectively escalated from optional to mandatory.

The intended goal is to prevent the configuration/installation error to lurk silently until an end user finally submit a workflow. Instead, it should fail loudly on the sysadmin upon server start-up.

**Comment by tcompa** [comment]
This issue concerns parsl, and not a generic backend.

At the moment the server cannot run if parsl package is installed (for whatever reason) in the current environment. This is not a design choice (having a certain external package in the environment makes it impossible to use the server).

--

Proposed solution 1, simply modify this block
```python
try:
    from ._parsl import process_workflow as parsl_process_workflow

    _backends["parsl"] = parsl_process_workflow
except ModuleNotFoundError as e:
    _backend_errors["parsl"] = e
```
into
```python
try:
    from ._parsl import process_workflow as parsl_process_workflow

    _backends["parsl"] = parsl_process_workflow
except (ModuleNotFoundError, NotImplementedError) as e:
    _backend_errors["parsl"] = e
```

Proposed solution 2, replace the original try/except block (**for parsl only**) with:
```python
error = NotImplementedError("some deprecation message")
_backend_errors["parsl"] = e
```

**Comment by tcompa** [comment]
Note:
Since `get_process_workflow` is only called when a workflow-apply endpoint is called, the check is not done on startup.

**Comment by tcompa** [comment]
Also note:
This error takes place in a background task, and then it is currently "hidden" in the server logs.

**Comment by tcompa** [comment]
To close this issue without too much effort, let's simply not collect the parsl backend at all.

I'm not 100% sure on whether we should just remove all parsl-backend code (and tests) from our current codebase (it is deprecated and out of date, but it may be useful to look at it in the future if we ever want to implement a new parsl-based backend), but for sure there is no need to load it (possibly raising/handling errors).

**Comment by jluethi** [comment]
Sounds good. In that case, let's leave it there but not being loaded for the time being.
